calcom/cal.com
# feat: SSO for orgs

## What does this PR do?

This PR 
- Allows Orgs users in Cloud to use SAML/SSO & OIDC for siging in.
- Allows Auto-create of user account and link to org based on the email domain that matches with `OrganizationSettings.orgAutoAcceptEmail` for both identityProvider (Okta, etc.) initiated login and ServiceProvider (cal.com) initiated login.


 Loom Video: https://www.loom.com/share/e5c4b4f3090146c1bd66effc76083fd0

## Requirement/Documentation


- NA

## Type of change



- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How should this be tested?



- Are there environment variables that should be set?
set `ORGANIZATIONS_ENABLED` to `true`
test for hosted cal

- Essential to set up: https://cal.com/docs/introduction/quick-start/self-hosting/sso-setup#setting-up-saml-login

- Flows to be tested
1. Sign in via idP, confirm that the account is created if domain matches the value `OrganizationSettings.orgAutoAcceptEmail` for a verified organization, and not otherwise
2. Sign in via SP (cal.com login page), confirm that the account is created if domain matches the value `OrganizationSettings.orgAutoAcceptEmail` for a verified organization, and not otherwise
3. Be able to login normally with a account created above (1, 2) with SAML without issues once they are created.


## Mandatory Tasks


## Checklist
